### ISSUE TITLE
Version solving errors are hard to parse

### ISSUE THREAD

**Issue by gaaclarke** [dart-lang/pub#3363]

**Comment by vsmenon** [comment]
Actually, not sure I triaged this correctly.  I missed these are SDK packages.  Are those handled differently?

**Comment by gaaclarke** [comment]
The pubspec.yaml is in the dart sdk itself: https://github.com/dart-lang/sdk/blob/main/pkg/kernel/pubspec.yaml

The big problem is that it hard to read because there is no delineation between package names and pubspec concepts and English.  For example, I didn't author the pubspec so the idea that `expect` is a package name wasn't even on my radar.  I still have a hard time understanding what it says.
